src/api/formatters.ts

```diff
@@ -70,12 +71,26 @@ export async function initializeFormatters({
     await initializeFormatter(stdout, 'stdout', configuration.stdout)
   )
 
-  for (const [target, type] of Object.entries(configuration.files)) {
-    const stream: IFormatterStream = fs.createWriteStream(null, {
-      fd: await fs.open(path.resolve(cwd, target), 'w'),
-    })
-    formatters.push(await initializeFormatter(stream, target, type))
-  }
+  const streamPromises: Promise<void>[] = []
+
+  Object.entries(configuration.files).forEach(([target, type]) => {
+    streamPromises.push(
+      (async (target, type) => {
+        const absoluteTarget = path.resolve(cwd, target)
+
+        if (!(await fs.exists(absoluteTarget))) {
```

Comment:
`fs.exists` is deprecated, and `mkdirp` is going to be [doing this check anyway](https://github.com/isaacs/node-mkdirp#implementation), so I think it would make more sense to just call `mkdirp` unconditionally.

However I do think we should wrap in a try/catch to be safe, and if something goes wrong do e.g. `logger.warn('Failed to ensure directory for formatter target exists')` and continue.

Reply:
Requested changes applied.